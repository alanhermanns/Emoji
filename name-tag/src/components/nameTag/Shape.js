import React from 'react';
import PropTypes from 'prop-types';

const Shape = ({ text, color, backgroundColor }) => (
  <div style={{ color, backgroundColor, width: '20vw', height: '20vh' }}>
    {text}
  </div>
);

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
};

export default Shape;
