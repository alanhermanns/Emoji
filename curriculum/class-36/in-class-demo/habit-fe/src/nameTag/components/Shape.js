import React from 'react';
import { useNameTagText, useNameTagColor, useNameTagBackgroundColor } from '../hooks/nameTag';

const Shape = () => {
  const text = useNameTagText();
  const color = useNameTagColor();
  const backgroundColor = useNameTagBackgroundColor();

  return (
    <div style={{ color, backgroundColor, width: '50vw', height: '50vh' }}>
      {text}
    </div>
  );
};

export default Shape;
