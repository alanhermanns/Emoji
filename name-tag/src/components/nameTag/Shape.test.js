import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape component', () => {
  it('renders Shape', () => {
    const wrapper = shallow(<Shape
      text="hi"
      color="#FF0000"
      backgroundColor="#00FF00" />);
    expect(wrapper).toMatchSnapshot();
  });
});
