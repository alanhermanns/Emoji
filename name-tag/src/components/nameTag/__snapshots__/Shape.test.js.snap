// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders Shape 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#00FF00",
      "color": "#FF0000",
      "height": "20vh",
      "width": "20vw",
    }
  }
>
  hi
</div>
`;
